### PR TITLE
HOOT: various imps & fixes part 14

### PR DESCRIPTION
**Pull request by Arcasias**
Part 1: https://github.com/odoo/odoo/pull/152930
Part 2: https://github.com/odoo/odoo/pull/153018
Part 3: https://github.com/odoo/odoo/pull/153023
Part 4: https://github.com/odoo/odoo/pull/153203
Part 5: https://github.com/odoo/odoo/pull/153425
Part 6: https://github.com/odoo/odoo/pull/153700
Part 7: https://github.com/odoo/odoo/pull/154054
Part 8: https://github.com/odoo/odoo/pull/154579
Part 9: https://github.com/odoo/odoo/pull/155073
Part 10: https://github.com/odoo/odoo/pull/155639
Part 11: https://github.com/odoo/odoo/pull/156255 / https://github.com/odoo/enterprise/pull/58135
Part 12: https://github.com/odoo/odoo/pull/156869
Part 13: https://github.com/odoo/odoo/pull/158384 / https://github.com/odoo/enterprise/pull/59019

Enterprise: https://github.com/odoo/enterprise/pull/59291

This pull requests brings various improvements and fixes to Hoot and the Odoo unit test ecosystem. See the different commit messages for more details.

Note: these changes are made in stable to avoid having to support multiple versions of the HOOT API. As such, these changes are intended to be strictly limited to unit tests as to not put the rest of the code base at risk.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/tests/chatter/web/follower.test.js [hunk 1]
```diff
@@ -186,9 +186,9 @@ test("removing a follower should reload form view", async function () {
         res_id: threadId,
         res_model: "res.partner",
     });
-    onRpc((route, args) => {
```
[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/mail/static/tests/chatter/web/follower.test.js:189]: form onRpc("web_read", ({ args, model }) not explained in commit message

File: addons/web/static/tests/_framework/mock_server/mock_server.js [hunk 2]
```diff
@@ -263,16 +276,17 @@ export class MockServer {
         // Set default routes
         this.onRpc("/web/action/load", this.mockActionLoad);
         this.onRpc("/web/bundle", this.mockBundle, { pure: true });
-        this.onRpc("/web/dataset/call_kw", this.mockCallKw);
-        this.onRpc("/web/dataset/call_button", this.mockCallKw);
+        this.onRpc("/web/dataset/call_kw", this.mockCallKw, { alwaysReturns: true });
```
Comment (Polymorphe57) [inline addons/web/static/tests/_framework/mock_server/mock_server.js:280]: why?

File: addons/mail/static/tests/chatter/web/follower.test.js [hunk 3]
```diff
@@ -186,9 +186,9 @@ test("removing a follower should reload form view", async function () {
...
-        if (route === "/web/dataset/call_kw/res.partner/web_read") {
-            step(`read ${args.args[0][0]}`);
+    onRpc("web_read", ({ args, model }) => {
+        if (model === "res.partner") {
```
Comment (seb-odoo) [inline addons/mail/static/tests/chatter/web/follower.test.js:190]: Hello, would it be possible to have a better syntax for ORM methods?

`onOrm(modelName, methodName, ...)` or something like that?

It makes more sense to see the model first, and it's more convenient than catching all methods that have the same name and having to filter on model later.

**Comment by Polymorphe57** [comment]
@robodoo r+

**Comment by fdardenne** [comment]
@robodoo rebase-ff

**Pull request closed without merging**